pkg/logql/syntax/ast.go

```diff
@@ -119,11 +169,6 @@ func IsMatchEqualFilterer(filterer log.LabelFilterer) bool {
 	}
 }
 
-// implicit holds default implementations
-type implicit struct{}
-
-func (implicit) logQLExpr() {}
```

Comment:
`implicit` is replaced by the explicit `isExpr()` function definition on each individual expression.